File: update.py
import requests
import json
import re
from datetime import datetime

def get_prices():
    res = requests.get("https://polttoaine.net", headers={"User-Agent": "Mozilla/5.0"})
    res.encoding = "latin-1"
    html = res.text

    match = re.search(r'Eilisen keskihinnat[\s\S]*?<tr[^>]*>([\s\S]*?)<\/tr>', html)
    if not match:
        raise Exception("Eilisen keskihinnat -riviä ei löytynyt")

    row = match.group(1)
    prices = re.findall(r'<td[^>]*class="Hinnat"[^>]*>(.*?)<\/td>', row)

    if len(prices) < 3:
        raise Exception(f"Liian vähän hintoja: {len(prices)}")

    return {
        "95E10": prices[0].strip(),
        "98E5": prices[1].strip(),
        "Diesel": prices[2].strip()
    }

def main():
    prices = get_prices()
    data = {
        "paivitetty": datetime.utcnow().isoformat(),
        "lahde": "polttoaine.net",
        "eilisen_keskihinnat": prices
    }
    with open("data.json", "w") as f:
        json.dump(data, f, indent=2)
    print("Päivitetty:", data)

main()

File: update.js
const fs = require("fs");

async function getPrices() {
    const res = await fetch("https://polttoaine.net");
    const buffer = await res.arrayBuffer();
    const html = new TextDecoder("latin1").decode(buffer);

    const match = html.match(/Eilisen keskihinnat[\s\S]*?<tr[^>]*>([\s\S]*?)<\/tr>/);
    if (!match) throw new Error("Eilisen keskihinnat -riviä ei löytynyt");

    const row = match[1];
    const prices = [...row.matchAll(/<td[^>]*class="Hinnat"[^>]*>(.*?)<\/td>/g)]
        .map(m => m[1].trim());

    if (prices.length < 3) throw new Error(`Liian vähän hintoja löytyi: ${prices.length}`);

    return {
        "95E10": prices[0],
        "98E5": prices[1],
        "Diesel": prices[2]
    };
}

async function main() {
    try {
        const prices = await getPrices();
        const data = {
            paivitetty: new Date().toISOString(),
            lahde: "polttoaine.net",
            eilisen_keskihinnat: prices
        };
        fs.writeFileSync("data.json", JSON.stringify(data, null, 2));
        console.log("Päivitetty:", data);
    } catch (err) {
        console.error("Virhe:", err.message);
        process.exit(1);
    }
}

main();
